### ISSUE TITLE
[PHP] Replacing `TRUE` with `true` strangely alters findings

### ISSUE THREAD

**Issue by IagoAbal** [semgrep/semgrep#5880]
https://semgrep.dev/s/wLPP produces two findings (one unintended), whereas https://semgrep.dev/s/xLJg produces just one finding (the one we want).

Note that in PHP `true` and `false` are case _insensitive_.

**Comment by r2c-demo** [comment]
This issue is synced in Linear at https://linear.app/r2c/issue/PA-1737/[php]-replacing-true-with-true-strangely-alters-findings. _Note_: this link is for r2c use only and is not accessible publicly.

**Comment by erwanlr** [comment]
Such behaviour happens with basically any uppercased keywords, for example:
- TRUE
- FALSE
- INT
- NULL

**Comment by Sjord** [comment]
This was due to a difference between pfff and tree-sitter. Pfff can't handle uppercase true, so this causes the program to be parsed with tree-sitter. Tree-sitter parses `die()` as `die()`, pfff as `exit()`, and these don't match. I made a PR to make tree-sitter parse `die()` as `exit()` too, solving the difference. Another possible improvement is to add string.lowercase to the [pfff PHP lexer](https://github.com/returntocorp/pfff/blob/1985f9975ef5ee4340cac61cefea28a53c89a73e/lang_php/parsing/lexer_php.mll#L560), so pfff can also parse uppercase true.
